### PR TITLE
feat: Get app validations

### PR DESCRIPTION
**Pull request by Aradhya-Tripathi**
## Summary

- Split the monolithic `app_validator.py` into a `pilot/core/app_validator/` package: `RepoStructureCheck`, `SyntaxCheck`, `DependencyDeclarationsCheck`, `ImportCheck`, orchestrated by a slimmed-down `Validator` that just runs a list of checks — new checks plug in without touching the others.

**Import validation (`ImportCheck`)**
- Installs the app into a throwaway venv and statically resolves every import it makes — no app code ever executes (stat-based `ModuleResolver` first, `find_spec` fallback only for confirmation).
- AST-based import collection handles relative imports (including detecting ones that climb above the app's own package), skips anything inside `try/except` (optional deps) and `if TYPE_CHECKING:` blocks, and excludes test files.
- Errors show the exact `file:line` each unresolved import came from, not just a bare module name.
- Resolves correctly regardless of PyPI dist-name vs. import-name mismatches (e.g. `beautifulsoup4` → `bs4`).

**Dependency declarations (`DependencyDeclarationsCheck`)**
- `pyproject.toml`'s `[tool.bench.frappe-dependencies]` is the source of truth for required frappe apps; cross-checked against `hooks.py`'s `required_apps`.
- Hardened against malformed pyproject shapes (wrong type instead of a list/table) and org-prefixed entries (`"someorg/app"`, not just `"frappe/app"`).

**Dependency-aware `get-app`**
- Resolves an app's dependencies via the marketplace registry and installs whatever's missing, in correct order, before installing the app itself.
- Falls back to the app's own `pyproject.toml` declarations when it's not in the registry, failing fast with a clear message naming exactly what's missing — never silently proceeding into a confusing later failure.
- Already-installed dependencies are detected up front, so the registry's full transitive resolution is skipped entirely when there's nothing to do (avoids needless registry rebuilds/subprocess spawns).
- Dependency installs are install-only — they don't re-run their own validation or dependency resolution; only the app the user actually asked for gets fully validated.
- `ImportCheck`'s tmp env now also installs the app's bench-resident required dependencies, so cross-app imports (e.g. `india_compliance` → `erpnext`) resolve correctly during validation instead of failing as missing.

**Safety fixes**
- `get-app frappe` is blocked outright — frappe is the base framework, not installable via `get-app`.
- A validation or dependency-resolution failure only rolls back the clone if *this run* created it — re-running `get-app` on an already-installed app can never wipe it out.

## Test plan

- [x] ~18 new/updated tests in `test_app_validator.py`: relative-import edge cases, try/except and `TYPE_CHECKING` guards, dist-name mismatches, dependency-declaration pass/fail cases
- [x] Real tmp-env integration tests (actual venv + pip install) alongside fast unit-level ones
- [x] Verified end-to-end against real benches with frappe/erpnext/india_compliance

### CODE REVIEW

File: pilot/core/app_validator/imports.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        self, nodes: Iterable[ast.AST]
+    ) -> Iterator[ast.Import | ast.ImportFrom]:
+        """Imports that must resolve at runtime — skips imports guarded by
+        try/except ImportError and `if TYPE_CHECKING:` blocks, which apps use
+        for genuinely optional dependencies."""
+        for node in nodes:
+            if isinstance(node, (ast.Import, ast.ImportFrom)):
+                yield node
+            elif isinstance(node, ast.Try) and self._handles_import_error(node):
+                continue
```
[the remark this answers is not in the document]
Comment (Aradhya-Tripathi) [reply pilot/core/app_validator/imports.py]: We will be skipping imports in try:except completely hoping the user knows what they are doing in that case.

File: pilot/core/app_validator/utils/tmp_env.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            raise AppValidationError(
+                f"Failed to install frappe into the validation env:\n{exc.message}"
+            )
+        return self
+
+    def install_app(self, app: "App") -> None:
+        try:
+            self._pip_install([app.path])
+        except CommandError as exc:
+            raise AppValidationError(f"'{app.config.name}' failed to install:\n{exc.message}")
```
[the remark this answers is not in the document]
Comment (Aradhya-Tripathi) [reply pilot/core/app_validator/utils/tmp_env.py]: This isn't in our threat model's scope.

Don't let the app execute any code wasn't the case before and won't be the case after this PR either.
